File: MiniDOMFinanziaria/Finanziaria/Convenzioni/Convenzioni.vb
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Internals
Imports minidom.Finanziaria
Imports minidom.Anagrafica

Namespace Internals

    Public NotInheritable Class CConvenzioniClass
        Inherits CModulesClass(Of CQSPDConvenzione)

        Friend Sub New()
            MyBase.New("modCQSPDConvenzioni", GetType(CQSPDConvenzioniCursor), -1)
        End Sub

        Public Function GetConvenzioniPerProdotto(ByVal item As CCQSPDProdotto, Optional ByVal onlyValid As Boolean = False) As CCollection(Of CQSPDConvenzione)
            If (item Is Nothing) Then Throw New ArgumentNullException("item")
            Dim ret As New CCollection(Of CQSPDConvenzione)
            If (GetID(item) = 0) Then Return ret
            For Each c As CQSPDConvenzione In Me.LoadAll
                If (c.IDProdotto = GetID(item) AndAlso (Not onlyValid OrElse c.IsValid())) Then ret.Add(c)
            Next
            Return ret
        End Function

        Public Function GetConvenzioniPerAmministrazione(ByVal amm As CAzienda, Optional ByVal onlyValid As Boolean = False) As CCollection(Of CQSPDConvenzione)
            If (amm Is Nothing) Then Throw New ArgumentNullException("amministrazione")
            Dim ret As New CCollection(Of CQSPDConvenzione)
            If (GetID(amm) = 0) Then Return ret
            For Each c As CQSPDConvenzione In Me.LoadAll
                If (
                    (c.IDAmministrazione = GetID(amm) OrElse (amm.TipoRapporto = c.TipoRapporto)) AndAlso
                    (Not onlyValid OrElse c.IsValid())
                    ) Then ret.Add(c)
            Next
            Return ret
        End Function



    End Class
End Namespace

Partial Public Class Finanziaria



    Private Shared m_Convenzioni As CConvenzioniClass = Nothing

    Public Shared ReadOnly Property Convenzioni As CConvenzioniClass
        Get
            If (m_Convenzioni Is Nothing) Then m_Convenzioni = New CConvenzioniClass
            Return m_Convenzioni
        End Get
    End Property

End Class

File: MiniDOMFinanziaria/Finanziaria/Convenzioni/Convenzioni.cs
using System;
using minidom.internals;

namespace minidom
{
    namespace internals
    {
        public sealed class CConvenzioniClass : CModulesClass<Finanziaria.CQSPDConvenzione>
        {
            internal CConvenzioniClass() : base("modCQSPDConvenzioni", typeof(Finanziaria.CQSPDConvenzioniCursor), -1)
            {
            }

            public CCollection<Finanziaria.CQSPDConvenzione> GetConvenzioniPerProdotto(Finanziaria.CCQSPDProdotto item, bool onlyValid = false)
            {
                if (item is null)
                    throw new ArgumentNullException("item");
                var ret = new CCollection<Finanziaria.CQSPDConvenzione>();
                if (DBUtils.GetID(item) == 0)
                    return ret;
                foreach (Finanziaria.CQSPDConvenzione c in LoadAll())
                {
                    if (c.IDProdotto == DBUtils.GetID(item) && (!onlyValid || c.IsValid()))
                        ret.Add(c);
                }

                return ret;
            }

            public CCollection<Finanziaria.CQSPDConvenzione> GetConvenzioniPerAmministrazione(Anagrafica.CAzienda amm, bool onlyValid = false)
            {
                if (amm is null)
                    throw new ArgumentNullException("amministrazione");
                var ret = new CCollection<Finanziaria.CQSPDConvenzione>();
                if (DBUtils.GetID(amm) == 0)
                    return ret;
                foreach (Finanziaria.CQSPDConvenzione c in LoadAll())
                {
                    if ((c.IDAmministrazione == DBUtils.GetID(amm) || (amm.TipoRapporto ?? "") == (c.TipoRapporto ?? "")) && (!onlyValid || c.IsValid()))
                        ret.Add(c);
                }

                return ret;
            }
        }
    }

    public partial class Finanziaria
    {
        private static CConvenzioniClass m_Convenzioni = null;

        public static CConvenzioniClass Convenzioni
        {
            get
            {
                if (m_Convenzioni is null)
                    m_Convenzioni = new CConvenzioniClass();
                return m_Convenzioni;
            }
        }
    }
}